oauth2client/_transport/httplib2.py

```diff
@@ -157,7 +170,18 @@ def request(http, uri, method='GET', body=None, headers=None, **kwargs):
     Returns:
         An instance of :class:`_ResponseWrapper`.
     """
-    response, data = http.request(
-        uri, method=method, body=body, headers=headers, **kwargs)
+
+    # Httplib2 doesn't let you set the timeout when calling request, however,
+    # it is a public attribute and httplib2 is not thread-safe, so just change
+    # it.
+    original_timeout = http.timeout
+    if timeout is not None:
+        http.timeout = timeout
+
+    try:
+        response, data = http.request(
+            uri, method=method, body=body, headers=headers, **kwargs)
+    finally:
+        http.timeout = original_timeout
```

Comment:
Sure, but there isn't a context manager specifically for this (other than `mock.patch` which shouldn't be used in real code. :P)